R/tspa_plot_revised.R

```diff
@@ -0,0 +1,123 @@
+tspa_plot <- function(tspa_fit,
+                      ask = FALSE,
+                      title = NULL,
+                      label_x = NULL,
+                      label_y = NULL,
+                      abbreviation = TRUE,
+                      fscore_type = NULL,
+                      ...) {
+
+  fit_pars <- lavaan::parameterestimates(tspa_fit) # parameter estimates
+  regression_fit <- fit_pars[which(fit_pars$op == "~"),]
+  latent_dv <- unique(c(unname(t(regression_fit["lhs"])))) # Extract DV
+  latent_iv <- unique(c(unname(t(regression_fit["rhs"])))) # Extract IV
+  fscores_func <- lavaan::lavInspect(tspa_fit, what = "data") # factor scores from `get_fs`
+  fscores_est <- lavaan::lavPredict(tspa_fit) # factor scores from estimation using `lavPredict`
+
+    # Type of scatterplot
+    if (is.null(fscore_type)) {
+      fscores <- fscores_est
+    } else {
+      fscores <- fscore_func
+    }
+
+    # Helper function for scatter plot
+    # Comment: put it outside the main function
```

Comment:
As mentioned, have the function separate if possible.

Reply:
Solved.